src/workerd/util/mimetype.c++

```diff
@@ -253,39 +257,47 @@ kj::String MimeType::essence() const {
 }
 
 kj::String MimeType::paramsToString() const {
-  auto str = kj::strTree();
+  StringBuffer<MAX_MIMETYPE_LEN> buffer;
+  paramsToString(buffer);
+  return buffer.toString();
+}
+
+template<size_t Capacity>
+void MimeType::paramsToString(StringBuffer<Capacity>& buffer) const {
   bool first = true;
   for (auto& param : params()) {
-    str = kj::strTree(kj::mv(str), first ? "" : ";", param.key, "=");
+    buffer.append(first ? "" : ";", param.key, "="_kj);
     first = false;
     if (param.value.size() == 0) {
-      str = kj::strTree(kj::mv(str), "\"\"");
+      buffer.append("\"\"");
     } else if (hasInvalidCodepoints(param.value, isTokenChar)) {
       auto view = param.value.asPtr();
-      str = kj::strTree(kj::mv(str), "\"");
+      buffer.append("\"");
       while (view.size() > 0) {
         KJ_IF_MAYBE(pos, view.findFirst('"')) {
-          str = kj::strTree(kj::mv(str), view.slice(0, *pos), "\\\"");
+          buffer.append(view.slice(0, *pos), "\\\"");
           view = view.slice(*pos + 1);
         } else {
-          str = kj::strTree(kj::mv(str), view);
+          buffer.append(view);
           view = view.slice(view.size());
         }
       }
-      str = kj::strTree(kj::mv(str), "\"");
+      buffer.append("\"");
     } else {
-      str = kj::strTree(kj::mv(str), param.value);
+      buffer.append(param.value);
     }
   }
-  return str.flatten();
 }
 
 kj::String MimeType::toString() const {
-  auto str = kj::strTree(type(), "/", subtype());
+  StringBuffer<MAX_MIMETYPE_LEN> buffer;
+  buffer.append(type(), "/", subtype());
   if (params_.size() > 0) {
-    str = kj::strTree(kj::mv(str), ";", paramsToString());
+    buffer.append(";");
+    paramsToString(buffer);
   }
-  return str.flatten();
+  KJ_ASSERT(!buffer.hasOverflow());
```

Comment:
I wonder if we need this assert at all? @jasnell

Reply:
Probably not.

Reply:
But I prefer @kentonv's idea that if we do overflow we should switch to a heap allocated buffer instead.